core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -2615,7 +2616,13 @@ class ReplicaManagerTest {
 
   @Test
   def testStopReplicaWithDeletePartitionAndExistingPartitionAndNewerLeaderEpochAndIOException(): Unit = {
-    testStopReplicaWithExistingPartition(2, true, true, Errors.KAFKA_STORAGE_ERROR)
+    // 1. Even though we are trying to trigger an IOException by deleting the underlying log directory,
+    // the foreground deletion of a replica no longer reads from the underlying directory, and thus
+    // should trigger no exceptions. It means the stopReplica operation running the foreground
+    // deletion should be able to finish without any errors.
+    // 2. During the background deletion, a FileSystemException will be triggered when we try
+    // to delete the log segments, which is then converted to a KafkaStorageException.
+    testStopReplicaWithExistingPartition(2, true, true, Errors.NONE, true)
```

Comment:
Since all tests now set expectedOutput to NONE, could we just remove that param?

Reply:
There are still tests that expects an error, e.g. https://github.com/apache/kafka/blob/f809fccd0246b8263f341634c2e38749b132ef06/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala#L2630